File: game/src/test/java/integration/TestUtilsConfiguration.java
package integration;

import com.matag.game.cardinstance.CardInstanceFactory;
import com.matag.cards.Cards;
import com.matag.game.player.PlayerFactory;
import com.matag.game.stack.SpellStack;
import com.matag.game.status.GameStatusFactory;
import com.matag.game.turn.Turn;
import integration.cardinstance.CardsTestConfiguration;
import integration.turn.player.PlayerTestConfiguration;
import integration.turn.status.GameStatusTestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;

@Configuration
@Import({CardsTestConfiguration.class, GameStatusTestConfiguration.class, PlayerTestConfiguration.class, Turn.class, SpellStack.class})
public class TestUtilsConfiguration {
  @Bean
  public TestUtils testUtils(GameStatusFactory gameStatusFactory, PlayerFactory playerFactory, CardInstanceFactory cardInstanceFactory, Cards cards) {
    return new TestUtils(gameStatusFactory, playerFactory, cardInstanceFactory, cards);
  }

}

File: src/test/kotlin/integration/TestUtilsConfiguration.kt
package integration

import com.matag.cards.Cards
import com.matag.game.cardinstance.CardInstanceFactory
import com.matag.game.init.NoInitService
import com.matag.game.player.PlayerFactory
import com.matag.game.status.GameStatusFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class TestUtilsConfiguration {
    @Bean
    open fun testUtils(
        gameStatusFactory: GameStatusFactory,
        playerFactory: PlayerFactory,
        cardInstanceFactory: CardInstanceFactory,
        cards: Cards
    ): TestUtils {
        return TestUtils(gameStatusFactory, playerFactory, cardInstanceFactory, cards)
    }

    @Bean
    open fun noInitSevice() = NoInitService()
}